layouts/partials/tw/nav.html

```diff
@@ -1,137 +1,311 @@
+      <div class="space-y-1">
+        <span
+          class="
+            block h-[2px] w-5 origin-center rounded-full bg-black transition-transform ease-in-out
+          "
+        ></span>
+        <span
+          class="
+            block h-[2px] w-5 origin-center rounded-full bg-black transition-transform ease-in-out
+          "
+        ></span>
+        <span
+          class="
+            block h-[2px] w-3 origin-center rounded-full bg-black transition-transform ease-in-out
+          "
+        ></span>
+      </div>
+    </div>
+  </a>
+
+  <ul
+    class="
+      -top-2 right-1 z-10 ml-1 box-content w-0 flex-wrap gap-2 overflow-hidden bg-white shadow sm:relative sm:top-auto sm:flex sm:h-auto sm:w-auto sm:max-w-4xl sm:flex-1 sm:flex-nowrap sm:items-center sm:justify-evenly sm:rounded-none sm:border-none sm:bg-transparent sm:bg-none sm:px-16 sm:shadow-none sm:before:content-[] md:overflow-visible md:px-[6.75rem]
+    "
+    :class="toggle ? 'max-sm:absolute max-sm:border rounded border-s-5 flex w-auto h-auto px-4 pt-14 pb-10 sm:pt-0 sm:pb-0 sm:overflow-visible' : ''"
+    x-on="toggle"
+    :id="$id('dropdown-button')"
+  >
+    {{ range site.Menus.main }}
+      <li
+        class="flex h-11 flex-[1_0_45%] items-center text-sm leading-none sm:flex-[0_0_auto] md:justify-center"
+      >
         <a
-          href="/search/"
-          class="site-nav:item"
-          data-ga-category="search"
-          :data-ga-action="'modal:search:open'"
-          x-data
-          @click.prevent="$dispatch('modal:search:open')"
+          class="flex-[0_0_auto] font-bold uppercase hover:underline hover:underline-offset-2 sm:font-normal"
+          href="{{ .URL }}"
+          data-ga-category="item"
+          >{{ .Name }}</a
         >
-          <svg class="site-nav:svg-icon" aria-label="Search">
-            {{ partial "helper/svg.html" "search" }}
-          </svg>
-        </a>
+      </li>
+    {{ end }}
+    {{ range site.Menus.main_about }}
+      <li
+        class="
+          mt-4 flex-[0_0_100%] flex-wrap items-center text-sm leading-none 
+          sm:absolute sm:z-10 sm:h-11 sm:w-0 sm:overflow-hidden 
+          md:right-1 md:top-0 md:my-0 md:flex
+          md:flex-[0_0_auto] md:overflow-visible 
+        "
+        :class="toggle ? 'sm:block sm:right-[-4rem] md:right-[-4rem] sm:-top-6 sm:pt-2 sm:pb-3 sm:pr-8 sm:pl-4 sm:rounded-tr sm:bg-white sm:overflow-visible sm:w-48 sm:h-[auto] md:absolute md:h-auto md:top-[-1rem] md:pt-[15px] box-content sm:border-s-5 sm:border-r sm:border-t' : 'sm:overflow-hidden sm:w-0 sm:h-11'"
+      >
         <a
-          class="site-nav:donate-button"
-          href="/donate/"
-          data-ga-category="donate"
+          class="
+            sm:h block w-full flex-1 border-b py-4 font-bold uppercase hover:underline hover:underline-offset-2 sm:flex sm:w-full sm:items-center sm:capitalize md:py-0
+          "
+          :class="toggle ? 'sm:py-0 sm:h-11 md:justify-start' : 'md:justify-end'"
```

Comment:
Adding `md:justify-end` to the plain `class` fixes the flicker for me.

Reply:
This doesn't work for me because we don't have control over the cascade. `md:justify-end` overwrites `:class= … md:justify-start`

I've tried every combination I can think of to position the class but Alpine cannot override. The only solution I have come up with is an arbitrary class in tw.css https://github.com/spotlightpa/poor-richard/pull/422/commits/25ca397c1d6a1e82640113720c2f9a44b96adf1e

Reply:
Alpine has two ways to do `:class` bindings. One is a simple string, like you have here, but the other is an object with string keys and bool values. If you do the object way, Alpine will correctly turn off a class when you don't want it anymore. It would look like 

```html
:class="{
  'sm:py-0': toggle,
  'sm:h-11': toggle,
  'md:justify-start': toggle,
  'md:justify-end': ! toggle,
}"
```

Reply:
Here are the Alpine docs: https://alpinejs.dev/directives/bind#class-object-syntax

> This technique offers a unique advantage to other methods. When using object-syntax, Alpine will NOT preserve original classes applied to an element's class attribute.
> 
> For example, if you wanted to apply the "hidden" class to an element before Alpine loads, AND use Alpine to toggle its existence you can only achieve that behavior using object-syntax.

Basically by default, Alpine appends new classes to the DOM but leaves the initial ones from the HTML still there, so the only way to turn a class that is in the HTML off is with the object syntax.